openquake/hazardlib/site.py

```diff
@@ -75,13 +80,14 @@ def __str__(self):
         >>> loc = openquake.hazardlib.geo.point.Point(1, 2, 3)
         >>> str(Site(loc, 760.0, True, 100.0, 5.0))
         '<Location=<Latitude=2.000000, Longitude=1.000000, Depth=3.0000>, \
-Vs30=760.0000, Vs30Measured=True, Depth1.0km=100.0000, Depth2.5km=5.0000>'
+Vs30=760.0000, Vs30Measured=True, Depth1.0km=100.0000, Depth2.5km=5.0000, \
+Backarc=False>'
         """
         return (
             "<Location=%s, Vs30=%.4f, Vs30Measured=%r, Depth1.0km=%.4f, "
-            "Depth2.5km=%.4f>") % (
+            "Depth2.5km=%.4f, Backarc=%r>") % (
             self.location, self.vs30, self.vs30measured, self.z1pt0,
-            self.z2pt5)
+            self.z2pt5, self.backarc)
```

Comment:
IMO, this is more an issue of modelling than it is of the GMPE. In Si and Midorikawai it is true that they use the volcanic front and subduction zone to define the forearc from the location, but then the actual correction uses the forearc as a boolean term (https://github.com/gem/oq-hazardlib/blob/master/openquake/hazardlib/gsim/si_midorikawa_1999.py#L130). The fact that this is hard-coded into the GMPE makes Si & Midorikawa non-transferable to other regions (or at least non-transferable with the backarc term). It would have been better to have defined the backarc term already as a Boolean attribute of the site, and if necessary create a more general method (either here or in the toolkit) that can determine the backarc parameter from the location of the subduction and volcanic fronts. Also, I am seeing another subduction GMPE (Ghofrani & Atkinson, 2014) that uses backarc as a boolean, so I would go with this for now.